servers/rendering/renderer_rd/shaders/forward_clustered/scene_forward_clustered.glsl

```diff
@@ -836,7 +837,17 @@ vec4 fog_process(vec3 vertex) {
 		}
 	}
 
-	float fog_amount = 1.0 - exp(min(0.0, -length(vertex) * scene_data_block.data.fog_density));
+	float fog_amount = 0.0;
+
+	if (sc_use_depth_fog && scene_data_block.data.fog_mode == 1) {
```

Comment:
If using the specialization constant, you shouldn't need to check both `sc_use_depth_fog` and `scene_data_block.data.fog_mode` as `sc_use_depth_fog` should always be true when `scene_data_block.data.fog_mode` is true

Reply:
Also, it seems the specialization constant is never set for the Forward+ backend, so maybe that's why its not working? You have added it to the mobile renderer, but not the Forward+ renderer

Reply:
> Also, it seems the specialization constant is never set for the Forward+ backend, so maybe that's why its not working? You have added it to the mobile renderer, but not the Forward+ renderer

This part was a little more complex than I imagined, as the two renderers have very different codes, I couldn't understand how to configure the constant for the shaders. I tested this code by copying the light_filter above but I would need an environment, could you explain it better and tell me if there is something I am missing?

```cpp
sc.constant_id = SPEC_CONSTANT_USE_DEPTH_FOG;
sc.bool_value = environment_get_fog_mode() == RS::EnvironmentFogMode::ENV_FOG_MODE_DEPTH;

spec_constants.push_back(sc);
`